stdlib/public/Observation/Sources/Observation/Macros.swift

```diff
@@ -0,0 +1,26 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+#if SWIFT_OBSERVATION_MACROS
+#if compiler(>=5.8) && hasAttribute(attached)
```

Comment:
You shouldn't actually need the `hasAttribute` check here

Reply:
will remove